VideoLocker/src/main/java/org/edx/mobile/event/NewVersionAvailableEvent.java

```
@@ -0,0 +1,204 @@
+     * properties, if this hasn't been posted before. The sticky events will be queried for an
+     * existing report, and a new one will only be posted if it has more urgent information than the
+     * previous one. The events are posted and retained as sticky events in order to have a
+     * conveniently and semantically accessible session-based singleton of it to compare against,
+     * but this has the implication that they can't be removed from the event bus after consumption
+     * by the subscribers. To address this restriction, this class defined methods to mark instances
+     * as having being consumed, which can be used by subscribers for this purpose.
+     *
+     * If all the parameters are null or false, then it wouldn't be a valid event, and nothing would
+     * be posted on the event bus.
+     *
+     * @param newVersion        The version number of the latest release of the app.
+     * @param lastSupportedDate The last date on which the current version of the app will be
+     *                          supported.
+     * @param isUnsupported     Whether the current version is unsupported. This is based on whether
+     *                          we're getting HTTP 426 errors, and thus can't be inferred from the
+     *                          last supported date (the two properties may not be consistent with
+     *                          each other due to wrong local clock time or an inconsistency in the
+     *                          server configurations).
+     */
+    public static void post(@Nullable final Version newVersion,
+                            @Nullable final Date lastSupportedDate,
+                            final boolean isUnsupported) {
+        final NewVersionAvailableEvent event;
+        try {
+            event = new NewVersionAvailableEvent(newVersion, lastSupportedDate, isUnsupported);
+        } catch (IllegalArgumentException e) {
+            return;
```

Looks like you are hiding an error here. Could we perhaps have  a comment on why, and consider logging it as well (with Crashlytics report)?